providers/aws/connection/connection.go

```diff
@@ -118,32 +129,58 @@ func NewAwsConnection(id uint32, asset *inventory.Asset, conf *inventory.Config)
 		log.Info().Msg("no AWS region found, using us-east-1")
 		cfg.Region = "us-east-1" // in case the user has no region set, default to us-east-1
 	}
-	// gather information about the aws account
-	cfgCopy := cfg.Copy()
-	identity, err := CheckIam(cfgCopy)
+
+	c.Connection = plugin.NewConnection(id, asset)
+	c.Conf = conf
+	c.asset = asset
+	c.cfg = cfg
+	c.opts.profile = asset.Options["profile"]
+	c.opts.scope = asset.Options["scope"]
+	c.opts.options = asset.Options
+	if conf.Discover != nil {
+		c.Filters = parseOptsToFilters(conf.Discover.Filter)
+	}
+	return c, nil
+}
+
+func (c *AwsConnection) Hash() uint64 {
+	// generate hash of the config options used to generate this connection
+	// used to avoid verifying a client with the same options more than once
```

Comment:
```suggestion
	// generate hash of the config options used to to initialize this connection,
	// we use this to avoid verifying a client with the same options more than once
```